docs/framework/server-side-editor-api.md

```diff
@@ -0,0 +1,336 @@
+
+// Process each item in the document.
+for ( const item of items ) {
+    editor.model.change( writer => {
+        // Use _recordAttributeChanges to ensure the change is properly recorded as a suggestion.
+        tcEditing._recordAttributeChanges( () => {
+            let href = item.getAttribute( 'linkHref' );
+
+            // Only process text proxies (parts of text nodes) that have a `linkHref` attribute.
+            if ( item.is( 'textProxy' ) && href ) {
+                // Update the link URL, for example changing '/docs/' to '/documents/'.
+                href = href.replace( '/docs/', '/documents/' );
+
+                // Set the new attribute value, which will be recorded as a suggestion.
+                writer.setAttribute( 'linkHref', href, item );
+            }
+        } );
+    } );
+}
+```
+
+## Resolving comments
+
+{@link features/comments Comments} feature allows your users to have discussions on certain parts of your documents. You can use Comments feature API to implement interactions with comments with no need to open the editor itself.
+
+For example, here's how to resolve all comment threads in a given document:
+
+```js
+// Get all comment threads from the document.
+const threads = editor.plugins.get( 'CommentsRepository' ).getCommentThreads();
+
+// Resolve all open comment threads.
+for ( const thread of threads ) {
+    if ( !thread.isResolved ) {
+        thread.resolve();
+    }
+}
+```
+
+This code is particularly useful when you need to clean up a document before finalizing it. You might use it to automatically resolve old discussions, prepare documents for publication, or maintain a clean comment history in your content management system.
+
+## Working with revision history
+
+Use {@link features/revision-history Revision history} feature API to build more functional integration between your application and the document revisions data.
+
+### Saving revisions
+
+You can use Revision History API to save a new revision directly from your application back-end:
+
+```js
+// Save the current state as a new revision.
+editor.plugins.get( 'RevisionTracker' ).saveRevision( { name: 'New revision' } );
+```
+
+This can be used after you performed some changes to the document to save them as a new revision.
+
+You can also build an automated mechanism that will automatically create revisions in some time intervals, or based on other factors. It can be particularly useful when you need to create checkpoints for your documents to maintain an audit trail of content modifications.
+
+### Working with revision data
+
+For more advanced scenarios, you might need to work with different revisions of your document:
+
+```js
+// Get the revision management tools.
+const revisionHistory = editor.plugins.get( 'RevisionHistory' );
+const revisionTracker = editor.plugins.get( 'RevisionTracker' );
+
+// Get the first revision from history.
+const revision = revisionHistory.getRevisions()[ 0 ];
```

Comment:
Make sure it is `.getRevisions()[ 0 ]` not `.getRevisions( 0 )`. AFAIR they both return a different value 🤦.

Also, instead of using "first", let's use "oldest" or "latest". "First" is ambiguous -- first created, or first on the list?